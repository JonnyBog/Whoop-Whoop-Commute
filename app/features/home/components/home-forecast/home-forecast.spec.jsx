import React from 'react';
import renderer from 'react-test-renderer';

import HomeForecast from 'features/home/components/home-forecast/home-forecast';

describe('Components', () => {
  describe('Home Forecast', () => {
    it('should export', () => {
      expect(HomeForecast).toBeDefined();
    });

    let props;

    beforeEach(() => {
      props = {
        forecastSummary: 'test'
      };
    });

    afterEach(() => {
      props = null;
    });

    /**
     * createRenderTree
     * @param   {Object} updatedProps - defaults to props
     * @returns {JSX} component tree
     */
    function createRenderTree (updatedProps = props) {
      return renderer.create(
        <HomeForecast {...updatedProps} />
      ).toJSON();
    }

    describe('Appearance', () => {
      it('should render', () => {
        const tree = createRenderTree();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
