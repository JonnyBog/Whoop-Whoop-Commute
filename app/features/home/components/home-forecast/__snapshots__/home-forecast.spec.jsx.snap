// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Home Forecast Appearance should render 1`] = `
<div
  className="home-forecast__WeatherContainer-oz1efb-0 iHXbQ"
>
  <h2
    className="home-forecast__WeatherTitle-oz1efb-1 lnbJny"
  >
    Today's forecast:
  </h2>
  <p
    className="home-forecast__WeatherForecast-oz1efb-2 dZidty"
  >
    test
  </p>
</div>
`;
